dts: bindings: nxp,sctimer-pwm: add PWM period cell

gmarull opened the pull request:
The period cell will soon be required by the pwm_dt_spec facilities,
this patch adds support for it and updates boards accordingly.

Prep work for https://github.com/zephyrproject-rtos/zephyr/pull/44523

gmarull committed:
dts: bindings: nxp,sctimer-pwm: add PWM period cell

The period cell will soon be required by the pwm_dt_spec facilities,
this patch adds support for it.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arm: lpcxpresso55s69: add PWM period cell

Updated all PWM specs to include the period cell. Because all specs
refer to PWM driven LEDs, a period of 20 msec has been chosen, as most
other platforms do.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

anangl commented on dts/arm/nxp/nxp_lpc55S6x_common.dtsi:
```diff
@@ -390,7 +390,7 @@
 		prescaler = <2>;
 		label = "SC_TIMER";
-		#pwm-cells = <2>;
+		#pwm-cells = <3>;
```
The same should be done in nxp_rt5xx_common.dtsi:
https://github.com/zephyrproject-rtos/zephyr/blob/536fe6a948c9a546bb89dc73ab4adbd3b13b3e71/dts/arm/nxp/nxp_rt5xx_common.dtsi#L350
and in nxp_rt6xx_common.dtsi:
https://github.com/zephyrproject-rtos/zephyr/blob/536fe6a948c9a546bb89dc73ab4adbd3b13b3e71/dts/arm/nxp/nxp_rt6xx_common.dtsi#L302

gmarull replied:
fixed

anangl commented on boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts:
```diff
@@ -58,7 +58,7 @@
 		compatible = "pwm-leds";
 		red_pwm_led: red_pwm_led {
-			pwms = <&sc_timer 0 PWM_POLARITY_NORMAL>;
+			pwms = <&sc_timer 0 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
```
Also `mimxrt685_evk_cm33` needs to be updated: https://github.com/zephyrproject-rtos/zephyr/blob/dadcb3945a8eb6340262dc815225f61fada5e256/boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts#L73-L83

gmarull replied:
fixed

gmarull committed:
boards: arm: mimxrt685_evk: add PWM period cell

Updated all PWM specs to include the period cell. Because all specs
refer to PWM driven LEDs, a period of 20 msec has been chosen, as most
other platforms do.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

anangl commented on boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:
```diff
@@ -70,17 +70,17 @@
 		};
 		blue_pwm_led: blue_pwm_led {
-			pwms = <&sc_timer 6 PWM_POLARITY_NORMAL>;
+			pwms = <&sc_timer 6 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
 			label = "Blue PWM LED";
 			status = "okay";
 		};
 		red_pwm_led: red_pwm_led {
-			pwms = <&sc_timer 6 PWM_POLARITY_NORMAL>;
+			pwms = <&sc_timer 6 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
```
This seems to be wrong - both these LEDs use the same channel. I only spotted it now and I actually have no idea whether/how this should be fixed.

gmarull replied:
Hmm you're right. I'll check the board schematic and see if it can be fixed as part of this PR. Maybe @dleach02 or @danieldegrasse have some input here.

mmahadevan108 replied:
I confirmed and this is how it is connected on the board
